### ISSUE TITLE
New Encoder; update Requirements

### ISSUE THREAD

**Issue by ojaggy** [CorentinJ/Real-Time-Voice-Cloning#602]
Hey there,

first of all: thanks to all of you for your incredible work! 
I am currently training an encoder without the final relu layer, a model_hidden_size of 768 and 48 speakers per batch.
I've also adopted Corentin's new method of caluclating the similarity matrices as follows:

    sim_matrix = torch.zeros(speakers_per_batch, utterances_per_speaker,
                         speakers_per_batch).to(self.loss_device)
    eye = np.eye(speakers_per_batch, dtype=np.int)
    mask = np.where(1 - eye)
    sim_matrix[mask[0], :, mask[1]] = (embeds[mask[0]] * centroids_incl[mask[1]]).sum(dim=2)
    mask = np.where(eye)
    sim_matrix[mask[0], :, mask[1]] = (embeds * centroids_excl).sum(dim=2) 

![newplot](https://user-images.githubusercontent.com/67948001/99797426-846b1680-2b2f-11eb-8fbd-bc80e19895a9.png)

![newplot(1)](https://user-images.githubusercontent.com/67948001/99797439-8a60f780-2b2f-11eb-89d5-b87fd636f972.png)
![changed_umap_233500](https://user-images.githubusercontent.com/67948001/99797548-baa89600-2b2f-11eb-85aa-18772510a73a.png)

Because i have a RTX 3070 i wasn't able to use CUDA 10: The new Ampere Architecture is only supported by CUDA Version 11 + . But because the CUDA Versions are in fact backward compatible this isn't an issue. The only caveat is the used Pytorch Version: Since Pytorch doesn't support CUDA 11.1 one have to use CUDA 11.0 and Pytorch 1.7.

**Comment by blue-fish** [comment]
Closing due to inactivity. Reopen if you have more results to share. Thanks for noting that Pytorch 1.7 is compatible with the new 30xx series of GPUs.
